js/main.js

```diff
@@ -151,32 +151,34 @@ const getTypeTranslation = (type) => {
   return newTypeTranslation;
 };
 
-const checkFeatures = (featureElement, featureObject) => {
-  let isOnTheList = false;
-  for (let i = 0; i < featureObject.length; i++) {
-    if (featureElement.classList.contains(`popup__feature--${featureObject[i]}`)) {
-      isOnTheList = true;
-    }
-  }
-  return isOnTheList;
+const generateFeaturesItem = (featuresObjectItem) => {
+  let newFeaturesItem = document.createElement(`li`);
+  newFeaturesItem.classList.add(`popup__feature`);
+  newFeaturesItem.classList.add(`popup__feature--${featuresObjectItem}`);
+
+  return newFeaturesItem;
 };
 
-const generateFeaturesList = (featuresItem, featureObject) => {
-  const featuresElements = featuresItem.querySelectorAll(`.popup__feature`);
-  for (let i = 0; i < featuresElements.length; i++) {
-    if (checkFeatures(featuresElements[i], featureObject)) {
-      featuresElements[i].classList.add(`visually-hidden`);
-    }
+const createFeaturesFragment = (featuresObject) => {
+  let featuresFragment = document.createDocumentFragment();
+  for (let i of featuresObject) {
+    featuresFragment.appendChild(generateFeaturesItem(i));
   }
+  return featuresFragment;
+};
+
+const generatePhotoItem = (photoItem) => {
+  let newPhotoItem = cardTemplate.querySelector(`.popup__photo`).cloneNode(true);
+  newPhotoItem.src = photoItem;
+  return newPhotoItem;
 };
 
-const generatePhotosList = (photoItem, photosList) => {
-  photoItem.querySelector(`.popup__photo`).src = photosList[0];
-  for (let i = 1; i < photosList.length; i++) {
-    let newPhoto = cardTemplate.querySelector(`.popup__photo`).cloneNode(true);
-    newPhoto.src = photosList[i];
-    photoItem.appendChild(newPhoto);
+const createPhotoFragment = (featuresObject) => {
```

Comment:
Только тут копипасту надо будет потом поправить)
```suggestion
const createPhotoFragment = (photos) => {
```